domain/src/main/java/co/nimblehq/compose/crypto/domain/usecase/GetCoinDetailUseCase.kt

```diff
@@ -0,0 +1,15 @@
+package co.nimblehq.compose.crypto.domain.usecase
+
+import co.nimblehq.compose.crypto.domain.model.CoinDetail
+import co.nimblehq.compose.crypto.domain.repository.CoinRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class GetCoinDetailUseCase @Inject constructor(private val repository: CoinRepository) {
+
+    fun execute(coinId: String): Flow<CoinDetail> {
+        return repository.getCoinDetail(
+            coinId = coinId
+        )
```

Comment:
Optional: I think we can keep it one line as it has only one parameter 🤔 
```suggestion
        return repository.getCoinDetail(coinId = coinId)
```

Reply:
:+1